### PR TITLE
fix Issue 16400 - naked variadic C function emits broken prologue

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by thewilsonator** [comment]
This seems like this should target stable.

**Review by ghost — APPROVED** [review]

File: src/dmd/backend/cgelem.d [hunk 1]
```diff
@@ -5285,6 +5285,13 @@ private elem * elvalist(elem *e, goal_t goal)
 {
     assert(e.Eoper == OPva_start);
 
+    if (funcsym_p.ty() & mTYnaked)
+    {   // do not generate prolog
+        el_free(e);
+        e = el_long(TYint, 0);
```
Comment (UplinkCoder) [inline src/dmd/backend/cgelem.d:5291]: this is a 0 literal, no?

**Comment by UplinkCoder** [comment]
@NilsLankila I am not sure if this is ready yet?

File: src/dmd/backend/cgelem.d (hunk 1 above)
Comment (NilsLankila) [reply src/dmd/backend/cgelem.d:5291]: I'm 99% sure that this is a kind of noop, as when using the elxxx table there's always an elem expected as result, but later it's eliminated. I want to learn a bit the backend, a confirmation would be nice.

Also look at elinfo(), same trick already used.

**Comment by NilsLankila** [comment]
this is ready to me ;)

**Comment by thewilsonator** [comment]
This _codegen_ bug is still not targeting stable...

**Comment by NilsLankila** [comment]
@thewilsonator as last resort, you can rebase and force push in Walter's fork.

**Pull request merged**